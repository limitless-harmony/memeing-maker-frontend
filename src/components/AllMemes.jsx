import React, { Component } from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { calculateRem } from 'styles';
import { getMany } from 'actions/meme';
import MemeContainer from 'components/MemeContainer';

export class AllMemes extends Component {
  componentDidMount() {
    this.fetchMemes();
  }

  fetchMemes = async () => {
    const { actions, meta } = this.props;
    await actions.getMany(meta.nextPage || 1);
  };

  render() {
    const { memes, meta } = this.props;
    return (
      <StyledFeatured>
        {memes && memes.length ? (
          <>
            <SectionHeading>Available Memes</SectionHeading>
            <MemeContainer
              memes={memes}
              hasNextPage={meta.hasNextPage}
              fetchMore={this.fetchMemes}
            />
          </>
        ) : (
          <SectionHeading>No meme available yet!</SectionHeading>
        )}
      </StyledFeatured>
    );
  }
}

const StyledFeatured = styled.div`
  margin: ${calculateRem(10)} auto ${calculateRem(40)};
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  text-align: center;
`;

const SectionHeading = styled.div`
  font-size: ${calculateRem(18)};
  margin: ${calculateRem(16)} auto;
`;

const mapStateToProps = state => ({
  memes: state.meme.memes,
  meta: state.meme.meta,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({ getMany }, dispatch),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AllMemes);
