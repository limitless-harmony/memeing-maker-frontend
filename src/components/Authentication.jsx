import React, { Component } from 'react';
import styled from 'styled-components';

import { calculateRem } from 'styles';
import {
  FacebookColored,
  GoogleColored,
  LinkedinColored,
} from 'components/Icons';

class Authentication extends Component {
  handleLogin = strategy => {
    console.log(strategy);
  };

  loginWithGoogle = () => {
    return this.handleLogin('google');
  };

  loginWithFacebook = () => {
    return this.handleLogin('facebook');
  };

  loginWithLinkedIn = () => {
    return this.handleLogin('linkedIn');
  };

  render() {
    return (
      <StyledAuthentication>
        <TextSection>Sign up or log in</TextSection>
        <TextSection>to create, share, & discover</TextSection>
        <TextSection> our full library of meaningful memes.</TextSection>
        <ButtonSection>
          <Button onClick={this.loginWithGoogle}>
            <GoogleColored />
          </Button>
          <Button onClick={this.loginWithFacebook}>
            <FacebookColored />
          </Button>
          <Button onClick={this.loginWithLinkedIn}>
            <LinkedinColored />
          </Button>
        </ButtonSection>
      </StyledAuthentication>
    );
  }
}

const StyledAuthentication = styled.div`
  margin: ${calculateRem(-70)} auto auto;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  text-align: center;
`;

const TextSection = styled.div`
  align-items: center;
  font-size: ${calculateRem(18)};
`;

const ButtonSection = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  margin: ${calculateRem(20)} auto;
`;

const Button = styled.div`
  margin: 0 ${calculateRem(12)};
`;

export default Authentication;
