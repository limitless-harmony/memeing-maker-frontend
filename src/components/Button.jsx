import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import { calculateRem } from 'styles';
import { white, darken, green } from 'styles/colors';

const Button = ({ color, children, rounded, outline, fullWidth }) => {
  return (
    <StyledButton
      color={color}
      outline={outline}
      rounded={rounded}
      fullWidth={fullWidth}
    >
      {children}
    </StyledButton>
  );
};

export const StyledButton = styled.button`
  border: ${({ outline, color }) => (outline ? `2px solid ${color}` : 'none')};
  background-color: ${({ outline, color }) =>
    outline ? 'transparent' : color};
  color: ${({ outline, color }) => (outline ? color : white)};
  border-radius: ${({ rounded }) =>
    rounded ? calculateRem(4000) : calculateRem(8)};
  text-align: center;
  cursor: pointer;
  letter-spacing: 1px;
  width: ${({ fullWidth }) => (fullWidth ? '100%' : 'auto')};
  font-size: ${calculateRem(18)};
  margin: ${calculateRem(10)};
  padding: ${calculateRem(3)} ${calculateRem(30)};
  &:hover {
    background-color: ${({ color }) => darken(color, 0.05)};
    color: ${white};
  }
`;

Button.propTypes = {
  color: PropTypes.string,
  outline: PropTypes.bool,
  rounded: PropTypes.bool,
  fullWidth: PropTypes.bool,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
};

Button.defaultProps = {
  color: green,
  outline: false,
  rounded: false,
  fullWidth: false,
  children: 'Submit',
};

export default Button;
