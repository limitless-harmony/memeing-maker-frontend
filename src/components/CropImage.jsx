import React from 'react';
import styled from 'styled-components';
import Cropper from 'react-easy-crop';

const CropImage = ({
  image,
  crop,
  zoom,
  aspect,
  onCropChange,
  onCropComplete,
  onZoomChange,
}) => (
  <StyledCropImage>
    <Cropper
      image={image}
      crop={crop}
      zoom={zoom}
      aspect={aspect}
      onCropChange={onCropChange}
      onCropComplete={onCropComplete}
      onZoomChange={onZoomChange}
    />
  </StyledCropImage>
);

const StyledCropImage = styled.div`
  position: relative;
  width: 100%;
  height: 400px;
`;

export default CropImage;
