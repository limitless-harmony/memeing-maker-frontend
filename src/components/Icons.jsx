import { ReactComponent as Create } from 'assets/icons/create.svg';
import { ReactComponent as ShareButton } from 'assets/icons/share-button.svg';
import { ReactComponent as Facebook } from 'assets/icons/facebook.svg';
import { ReactComponent as Discover } from 'assets/icons/discover.svg';
import { ReactComponent as Ellipsis } from 'assets/icons/ellipsis.svg';
import { ReactComponent as AddImage } from 'assets/icons/add-image.svg';
import { ReactComponent as AddMemo } from 'assets/icons/add-memo.svg';
import { ReactComponent as FacebookColored } from 'assets/icons/facebook-colored.svg';
import { ReactComponent as FeelsReaction } from 'assets/icons/feels-reaction.svg';
import { ReactComponent as GoArrow } from 'assets/icons/go-arrow.svg';
import { ReactComponent as GoogleColored } from 'assets/icons/google-colored.svg';
import { ReactComponent as Instagram } from 'assets/icons/instagram.svg';
import { ReactComponent as LinkedinColored } from 'assets/icons/linkedin-colored.svg';
import { ReactComponent as Linkedin } from 'assets/icons/linkedin.svg';
import { ReactComponent as SocialShare } from 'assets/icons/social-share.svg';
import { ReactComponent as SocialLogins } from 'assets/icons/social-logins.svg';
import { ReactComponent as Submit } from 'assets/icons/submit.svg';
import { ReactComponent as ThanksReaction } from 'assets/icons/thanks-reaction.svg';
import { ReactComponent as ThoughtsReaction } from 'assets/icons/thoughts-reaction.svg';
import { ReactComponent as Twitter } from 'assets/icons/twitter.svg';
import { ReactComponent as WhatsApp } from 'assets/icons/whatsapp.svg';

export {
  Create,
  ShareButton,
  Facebook,
  Discover,
  Ellipsis,
  AddImage,
  AddMemo,
  FacebookColored,
  FeelsReaction,
  GoArrow,
  GoogleColored,
  Instagram,
  LinkedinColored,
  Linkedin,
  SocialShare,
  SocialLogins,
  Submit,
  ThanksReaction,
  ThoughtsReaction,
  Twitter,
  WhatsApp,
};
