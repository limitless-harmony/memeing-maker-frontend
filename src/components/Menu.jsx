import React, { Component } from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';

import { calculateRem, mobileWidth } from 'styles';
import { white, dark } from 'styles/colors';

export class Menu extends Component {
  handleClick = () => {};

  isPage = page => {
    const {
      location: { pathname },
    } = this.props;
    return pathname.includes(page);
  };

  render() {
    return (
      <StyledMenu>
        <MenuItem>
          Meme created by
          <Inlet>@username123</Inlet>
        </MenuItem>
        <MenuItem>
          Meme wall created by
          <Inlet>@username123</Inlet>
        </MenuItem>
        <MenuItem>Flag meme</MenuItem>
        <MenuItem>Mute meme</MenuItem>
        <MenuItem>Mute meme</MenuItem>
        <MenuItem>Mute player</MenuItem>
        <MenuItem>Add meme to wall</MenuItem>
        <MenuItem>Invite player to wall</MenuItem>
        <MenuItem>View player profile</MenuItem>
        <MenuItem>View your profile</MenuItem>
        <MenuItem>View rules of play</MenuItem>
        <MenuItem>Log out</MenuItem>
      </StyledMenu>
    );
  }
}

const StyledMenu = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 100vh;
  background: ${white};
  position: fixed;
  z-index: 5;
  width: 100%;
  max-width: ${calculateRem(mobileWidth)};
  box-sizing: border-box;
  left: 50%;
  top: 0%;
  transform: translateX(-50%);
  padding: ${calculateRem(60)} ${calculateRem(40)};
  font-size: ${calculateRem(15)};
  line-height: ${calculateRem(23)};
`;

const MenuItem = styled.div`
  cursor: pointer;
  color: ${dark};
  width: 100%;
`;
const Inlet = styled.div`
  padding: 0 ${calculateRem(40)};
  box-sizing: border-box;
`;

const mapStateToProps = state => ({
  location: state.router.location,
});

export default connect(mapStateToProps)(Menu);
