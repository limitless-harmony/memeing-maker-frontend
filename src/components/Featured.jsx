import React, { Component } from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { calculateRem } from 'styles';
import { getFeatured } from 'actions/meme';
import MemeCard from 'components/MemeCard';
import Button from 'components/Button';
import { buttonBorder, black } from 'styles/colors';

export class Featured extends Component {
  async componentDidMount() {
    const { actions } = this.props;
    await actions.getFeatured();
  }

  render() {
    const { memes } = this.props;
    return (
      <StyledFeatured>
        {memes.length ? (
          <>
            <SectionHeading>This Weekâ€™s Featured Memes</SectionHeading>
            <Container>
              {memes.map(meme => (
                // eslint-disable-next-line no-underscore-dangle
                <Meme key={meme._id}>
                  <MemeCard
                    small
                    src={meme.image}
                    topText={meme.topText}
                    bottomText={meme.bottomText}
                  />
                </Meme>
              ))}
            </Container>
            <Button color={buttonBorder} outline rounded>
              <ButtonText to="/login">More</ButtonText>
            </Button>
          </>
        ) : (
          <SectionHeading>No featured meme for this week!</SectionHeading>
        )}
      </StyledFeatured>
    );
  }
}

const StyledFeatured = styled.div`
  margin: ${calculateRem(10)} auto ${calculateRem(40)};
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  text-align: center;
`;

const Container = styled.div`
  margin: ${calculateRem(10)} auto ${calculateRem(40)};
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  width: 100%;
  text-align: center;
`;

const Meme = styled.div`
  width: 48.5%;
`;

const SectionHeading = styled.div`
  font-size: ${calculateRem(18)};
  margin: ${calculateRem(16)} auto;
`;

const ButtonText = styled(Link)`
  color: ${black};
  text-decoration: none;
`;

const mapStateToProps = state => ({
  memes: state.meme.featured,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({ getFeatured }, dispatch),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Featured);
