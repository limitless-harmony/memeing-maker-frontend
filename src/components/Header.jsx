import React, { Component } from 'react';
// import PropTypes from 'prop-types';
import styled from 'styled-components';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import { calculateRem, mobileWidth } from 'styles';
import { ShareButton, Discover, Ellipsis, Create } from 'components/Icons';
import { white } from 'styles/colors';
import { showModal } from 'actions/modal';
import toggleMenu from 'actions/menu';

export class Header extends Component {
  share = () => {
    const { isLoggedIn, actions } = this.props;
    if (!isLoggedIn) {
      return actions.showModal('auth');
    }
    return actions.showModal('share');
  };

  toggleMenu = () => {
    const { actions, menuStatus } = this.props;
    return actions.toggleMenu(!menuStatus);
  };

  render() {
    return (
      <StyledHeader>
        <NavLeft>
          <NavItem first>
            <Link to="/">
              <Discover />
            </Link>
          </NavItem>
          <NavItem>
            <Link to="/create">
              <Create />
            </Link>
          </NavItem>
          <NavItem>
            <ShareButton width="18" height="18" onClick={this.share} />
          </NavItem>
        </NavLeft>
        <NavRight>
          <NavItem last>
            <Ellipsis onClick={this.toggleMenu} />
          </NavItem>
        </NavRight>
      </StyledHeader>
    );
  }
}

export const StyledHeader = styled.header`
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: ${calculateRem(36)};
  padding: 0 ${calculateRem(14)};
  background: ${white};
  position: fixed;
  z-index: 10;
  width: 100%;
  max-width: ${calculateRem(mobileWidth)};
  box-sizing: border-box;
  left: 50%;
  top: 0%;
  transform: translateX(-50%);
`;

const NavItem = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  cursor: pointer;
  margin: ${({ first, last }) =>
    first
      ? `0 ${calculateRem(5)} 0 0`
      : last
      ? `0 0 0 ${calculateRem(5)}`
      : `0 ${calculateRem(5)}`};
`;

const NavSection = styled.div`
  display: flex;
  flex: 1;
`;

const NavLeft = styled(NavSection)`
  justify-content: flex-start;
`;

const NavRight = styled(NavSection)`
  justify-content: flex-end;
`;

Header.propTypes = {
  isLoggedIn: PropTypes.bool.isRequired,
  actions: PropTypes.shape({}).isRequired,
};

const mapStateToProps = state => ({
  menuStatus: state.menu.show,
  isLoggedIn: state.auth.isLoggedIn,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({ showModal, toggleMenu }, dispatch),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Header);
