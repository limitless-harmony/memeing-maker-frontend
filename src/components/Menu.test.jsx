import React from 'react';
import { shallow } from 'enzyme';

import { Menu } from 'components/Menu';

describe('Testing the Menu component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Menu />);
    expect(wrapper).toMatchSnapshot();
  });
});
