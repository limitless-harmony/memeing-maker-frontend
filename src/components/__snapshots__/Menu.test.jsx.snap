// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Menu component renders without crashing 1`] = `
<styled.div>
  <styled.div>
    Meme wall created by
    <styled.div>
      <Link
        replace={false}
        to="users/"
      >
        @username123
      </Link>
    </styled.div>
  </styled.div>
  <styled.div>
    Flag meme
  </styled.div>
  <styled.div>
    View player profile
  </styled.div>
  <styled.div>
    View your profile
  </styled.div>
  <styled.div>
    View rules of play
  </styled.div>
  <styled.div>
    Log out
  </styled.div>
</styled.div>
`;
